valohai/internals/global_state_loader.py

```diff
@@ -0,0 +1,227 @@
+    if os.path.isfile(config_path):
+        with open(config_path) as json_file:
+            inputs = json.load(json_file)
+
+    config_path = get_parameters_config_path()
+    if os.path.isfile(config_path):
+        with open(config_path) as json_file:
+            parameters = json.load(json_file)
+
+    return inputs, parameters
+
+
+def sift_cli_inputs(
+    args: argparse.Namespace, expected_keys: Set[str]
+) -> Dict[str, List[str]]:
+    """Sift inputs from all the command-line args
+
+    :param expected_keys: List of expected input names
+    """
+
+    result = {}
+    for name, values in vars(args).items():
+        # Filter out any inputs that we weren't expecting
+        if name not in expected_keys:
+            continue
+
+        if values is None:
+            continue
+
+        if not isinstance(values, list):
+            values = [values]
+
+        values = [v for v in values if v is not None]
+        if len(values) > 0:
+            result[name] = values
+    return result
+
+
+def sift_cli_parameters(
+    args: argparse.Namespace, expected_keys: Set[str]
+) -> Dict[str, Any]:
+    """Sift parameters from all the command-line args
+
+    :param expected_keys: List of expected parameter names
+    """
+
+    result = {}
+    for name, value in vars(args).items():
+        # Filter out any parameters that we weren't expecting
+        if name not in expected_keys:
+            continue
+
+        if value is not None:
+            result[name] = value
+    return result
+
+
+def sift_defaults(values: Dict[str, Any]) -> Dict[str, List[str]]:
+    """Returns the default values which user defined in .prepare()
+
+    Works for both inputs and parameters.
+
+    We sift from two alternatives (dict and nested dict):
+
+    valohai.prepare(
+        default_parameters={"my-param", 123},
+        default_inputs={"my-input", "https://lol.png"}
+        )
+    valohai.prepare(
+        default_parameters={"my-param", {"default": 123}},
+        default_inputs={"my-input", {"default": "https://lol.png"}}
+        )
+
+    :param values: Default values from the .prepare() method
+    """
+
+    result = {}
+    for key, value in values.items():
+        if isinstance(value, dict):
+            if "default" in value:
+                result[key] = value["default"]
+            else:
+                raise ValueError(f"No default value defined for {key}")
+        else:
+            result[key] = value
+    return result
+
+
+def convert_to_input_info(input: Union[str, List[str], Dict]) -> InputInfo:
+    """Converts inputs from different formats into an InputInfo
+
+    Inputs can be defined in either .prepare() or inputs.json
+
+    Inputs defined in .prepare() are a List[str] or str
+    Inputs defined in inputs.json are a Dict
+
+    Ultimately we want have all of them converted as InputInfo(s)
+
+    :param inputs: Single input as List[str] or Dict
+    """
+    if isinstance(input, (List, str)):
```

Comment:
While I guess using the typing List might work, this should probably be `list` the type.
```suggestion
    if isinstance(input, (list, str)):
```